### PR TITLE
 network: macvlan - add support to configure rx queue for broadcast / multicast

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/network/netdev/macvlan.c [hunk 1]
```diff
@@ -51,9 +52,53 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
...
+                m->bc_queue_length = UINT32_MAX;
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &m->bc_queue_length);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse BroadcastMulticastQueueLength=%s, ignoring assignment: %m", rvalue);
+                return 0;
+        }
```
Comment (yuwata) [inline src/network/netdev/macvlan.c:97]: Please refuse `UINT32_MAX`.

File: src/network/netdev/macvlan.c [hunk 2]
```diff
@@ -51,9 +52,53 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
...
+
         return 0;
 }
 
+
```
Comment (yuwata) [inline src/network/netdev/macvlan.c]: Hmm? Please drop this.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by ssahani** [comment]
Updated . Thanks for the review. ptal

File: src/basic/linux/if_link.h [hunk 3]
```diff
@@ -588,6 +588,8 @@ enum {
 	IFLA_MACVLAN_MACADDR,
 	IFLA_MACVLAN_MACADDR_DATA,
 	IFLA_MACVLAN_MACADDR_COUNT,
+    IFLA_MACVLAN_BC_QUEUE_LEN,
+    IFLA_MACVLAN_BC_QUEUE_LEN_USED,
```
Comment (bluca) [inline src/basic/linux/if_link.h]: nit: indentation seems off

File: src/network/netdev/macvlan.c [hunk 4]
```diff
@@ -51,6 +52,53 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
...
+        assert(rvalue);
+        assert(data);
+        assert(userdata);
+
+        if (isempty(rvalue)) {
+                m->bc_queue_length = UINT32_MAX;
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &m->bc_queue_length);
```
Comment (yuwata) [inline src/network/netdev/macvlan.c]: Please do not assign the value to `m->bc_queue_length` before checking its validity.
```
uint32_t v;
(snip)
r = safe_atou32(rvalue, &v);
if (r < 0) {
    ...;
}
if (v == UINT32_MAX) {
   ..;
}

m->bc_queue_length = v;

return 0;
```

File: src/basic/linux/if_link.h (hunk 3 above)
Comment (yuwata) [reply src/basic/linux/if_link.h]: Please try to use `src/basic/linux/update.sh`.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by ssahani** [comment]
updated . thanks for the review . ptal

File: src/network/netdev/macvlan.c (hunk 4 above)
Comment (ssahani) [reply src/network/netdev/macvlan.c]: By default `m->bc_queue_length = UINT32_MAX` . SO if safe_atou32 fails it still have that value it's not required to have a temp value.  Anyway UINT32_MAX not going to be assigned.

File: src/network/netdev/macvlan.c [hunk 5]
```diff
@@ -51,6 +52,56 @@ static int netdev_macvlan_fill_message_create(NetDev *netdev, Link *link, sd_net
...
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse BroadcastMulticastQueueLength=%s, ignoring assignment: %m", rvalue);
+                return 0;
+        }
+
+        if (v == UINT32_MAX) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Invalid BroadcastMulticastQueueLength=%s, ignoring assignment: %m", rvalue);
+        }
```
Comment (yuwata) [inline src/network/netdev/macvlan.c]: Please do not use `r` here; `r` -> `0`.

Also, missing `return 0;` in this block.

**Review by yuwata — CHANGES_REQUESTED** [review]
Mostly ok. But one more round please.

**Comment by ssahani** [comment]
updated thanks for the review ptal

**Pull request merged**